src/data/error.rs

```diff
@@ -79,3 +80,136 @@ impl<T, E> Recoverable for Result<T, (E, T)> {
         }
     }
 }
+
+pub enum CompileErrorLevel {
+    Warning,
+    RecoverableError,
+    FatalError,
+}
+
+pub trait NewCompileError {
+    // Idealy this should be an associated constant, but those are not allowed
+    // in trait objects
+    fn level(&self) -> CompileErrorLevel;
```

Comment:
~~This is not true: https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=3e1730a30bd1f73c3f7bc08e3ecdb84c~~ You meant `&dyn NewCompileError`, ignore me